alchemy/src/alchemy.ts

```diff
@@ -181,24 +182,34 @@ async function _alchemy(
     const root = new Scope({
       ...mergedOptions,
       appName,
+      phase,
+      password: mergedOptions?.password ?? process.env.ALCHEMY_PASSWORD,
+      telemetryClient,
+    });
+    const stage = new Scope({
+      ...mergedOptions,
+      scopeName:
+        mergedOptions?.stage ?? process.env.ALCHEMY_STAGE ?? process.env.USER,
+      parent: root,
+      appName,
       stage:
         mergedOptions?.stage ?? process.env.ALCHEMY_STAGE ?? process.env.USER,
       phase,
       password: mergedOptions?.password ?? process.env.ALCHEMY_PASSWORD,
       telemetryClient,
     });
     try {
-      Scope.storage.enterWith(root);
+      Scope.storage.enterWith(stage);
     } catch {
       // we are in Cloudflare Workers, we will emulate the enterWith behavior
       // see Scope.finalize for where we pop the global scope
-      Scope.globals.push(root);
+      Scope.globals.push(stage);
     }
     if (mergedOptions?.phase === "destroy") {
-      await destroy(root);
+      await destroy(stage);
       return process.exit(0);
     }
-    return root;
+    return stage;
```

Comment:
If we are now returning the stage, does that mean app.finalize will not be called on the actual root?

Reply:
it shouldn't matter because everything is also in stage scope, but we can return root scope, doesn't really matter